systemd/systemd.go

```diff
@@ -1065,18 +1082,53 @@ func (l Log) PID() string {
 	return "-"
 }
 
+type UnitLifetime int
+
+const (
+	Persistent UnitLifetime = iota
+	Transient
+)
+
 // MountUnitPath returns the path of a {,auto}mount unit
 func MountUnitPath(baseDir string) string {
 	escapedPath := EscapeUnitNamePath(baseDir)
 	return filepath.Join(dirs.SnapServicesDir, escapedPath+".mount")
 }
 
+// MountUnitPathWithLifetime returns the path of a {,auto}mount unit
+// created in the systemd directory suitable for the given unit lifetime
+func MountUnitPathWithLifetime(lifetime UnitLifetime, baseDir string) string {
```

Comment:
baseDir is not the clearest name?  wouldn't mountpointDir be a better name?

Reply:
is this going later to be used outside of systemd? because it's not in the current PR

Reply:
Done. This is not currently used, but it will be used by the `snapctl mount` command.